ivy_tests/test_ivy/test_functional/test_core/test_device.py

```diff
@@ -173,9 +173,8 @@ def test_as_native_dev(*, array_shape, dtype, as_variable, fw, call):
         else:
             assert ret == device
         # compilation test
-        if call is helpers.torch_call:
-            # pytorch scripting does not handle converting string to device
-            return
+        # pytorch scripting does not handle converting string to device
+        assume(not (call is helpers.torch_call))
```

Comment:
we should replace all `call is helpers.<name>_call` logic with `fw == "backend_name"` logic, both do exactly the same thing.